client/my-sites/stats/hooks/use-stats-purchases.tsx

```diff
@@ -121,24 +123,23 @@ export default function useStatsPurchases( siteId: number | null ) {
 	const hasLoadedSitePurchases = useSelector( hasLoadedSitePurchasesFromServer );
 
 	// Determine whether a product is owned.
-	// TODO we need to do plan check as well, because Stats products would be built into other plans.
-	const isFreeOwned = useMemo( () => {
-		return isProductOwned( sitePurchases, PRODUCT_JETPACK_STATS_FREE );
-	}, [ sitePurchases ] );
+	const isFreeOwned = useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, FEATURE_STATS_FREE );
+	} );
 
 	const isCommercialOwned = useMemo(
 		() => isCommercialPurchaseOwned( sitePurchases ),
 		[ sitePurchases ]
 	);
 
+	// Pay what you want
 	const isPWYWOwned = useMemo( () => {
 		return isProductOwned( sitePurchases, PRODUCT_JETPACK_STATS_PWYW_YEARLY );
 	}, [ sitePurchases ] );
```

Comment:
The `JETPACK_STATS_PWYW` product now set the `STATS_PAID` feature. Should we use it here?

Reply:
Interesting finding here, @macbre.

> The isPWYWOwned flag seems to control the display of the StatsPWYWOwnedNotice component. Can we just remove that flag and render the StatsPurchaseNotice component instead in client/my-sites/stats/stats-purchase/stats-purchase-notice.jsx (as we do for the commercial plans)?

I believe `StatsPurchaseNotice` cannot be used in this scenario because PWYW is a personal plan (non-commercial).

Perhaps the render condition for `StatsPWYWOwnedNotice` could be something like `isPaidOwned && ! isCommercialOwned` (indicating it has the `FEATURE_STATS_PAID` feature but not `FEATURE_STATS_COMMERCIAL`).

@Automattic/red, any thoughts?